Sources/WordPressKit/Services/AccountServiceRemoteREST.m

```diff
@@ -391,7 +391,16 @@ - (NSArray *)remoteBlogsFromJSONArray:(NSArray *)jsonBlogs
         // Exclude deleted sites from query result, since the app does not handle deleted sites properly.
         // I tried to use query arguments `site_visibility=visible` and `site_activity=active`, but neither excludes
         // deleted sites.
-        return !blog.isDeleted;
+        if (blog.isDeleted) {
+            return false;
+        }
+
+        // Exclude sites that are connected via Jetpack, but without an active Jetpack connection.
+        if (blog.jetpackConnection && !blog.jetpack) {
```

Comment:
What does WP.com do in the same scenario? Does it also remove the site from your account? What does "active Jetpack" mean? Can a site temporarily become inactive?

Reply:
`jetpack_connection = true` means the site was connected to WP.com via jetpack.

`jetpack = true` means WP.com can still communicate with the site via Jetpack (a.k.a, jetpack connection is active).

You typically get `jetpack_connection = true && jetpack = false` when the site was connected to WP.com a while ago, but its domain expired.

Reply:
WP.com does not show sites with `jetpack_connection = true && jetpack = false`, hence the fix.

There is more complicated logic on WP.com (see [here](https://github.com/Automattic/wp-calypso/blob/b57c069b819c61f4527c7737094ad2fd8ebed499/client/my-sites/sites/index.jsx#L50)) to decide whether to show a site or not. I didn't try to understand all of them. But this PR fixes the particular case where the site domain has expired.

Reply:
> Can a site temporarily become inactive?

I think that's entirely possible, like when the site is down, or for whatever reason that WP.com decides that it can't communicate with the site.

Reply:
I didn't follow the exact filtering logic on calypso. This is the code I referenced when looking into the API responses.

```javascript
// Sites connected through A4A plugin are listed on wordpress.com/sites even when Jetpack is deactivated.
export const isDisconnectedJetpackAndNotAtomic = ( site: SiteExcerptNetworkData ) => {
	return ! site?.is_wpcom_atomic && site?.jetpack_connection && ! site?.jetpack;
};
```

> I'd also suggest moving filtering (the entire logic inside wpkit_filter) to the app. It's not WordPressKit job to decide what app displays.

Agreed. I'll do that.